### DISCUSSION TITLE
How to change the background color of a 'Hint' on a textbox

### DISCUSSION THREAD

**Discussion by zonetecde** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3262] [Q&A]
Hello,

Is it possible to change the background color of the 'Hint' text so that it becomes transparent and does not create a contrast with the color of its parent?

Thank you

![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/56195432/63e6ca3a-9cc0-454b-9d64-6b8f95160115)

**Comment by zonetecde** [comment]
`materialDesign:HintAssist.Background="Transparent"` solved it!

**Comment by juusimaa** [comment]
Using latest version (5.1.0) I'm unable to se hint's background to transparent:

![image](https://github.com/user-attachments/assets/67cdc29a-fc9d-4f8d-9abd-cd35151626eb)

This is from an empty template app where `App.xaml` is updated according getting started guide where I try to set hint background to transparent.

**Comment by corvinsz** [comment]
@juusimaa without knowing exactly what you are trying to accomplish, setting the Background of your Window seems to be wrong and it's the main driver of your problem.

Have you considered following the [Getting Started Guide](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit?tab=readme-ov-file#getting-started) to fully leverage this library?
(You haven't set the Style on your Window Style="{StaticResource MaterialDesignWindow}")

**Comment by juusimaa** [comment]
I'm trying to get hint text background color to match in this case Grid's background color.This is from the Keboo template where I just added a TextBox and changed Grid's background color:

![image](https://github.com/user-attachments/assets/30966cd2-a68e-4383-92b7-451d2572ab26)

**Comment by nicolaihenriksen** [comment]
@juusimaa I think @corvinsz has a point about setting the custom background. Although you can definitely do it, you are not guaranteed that contrasts etc. work nicely in all cases.

With that said, even IF you were able to set the background to transparent, I don't believe this is what you want. It may not be obvious, but the white background in your example is (by the template) set to the default (theme-aware) window background color. It is set this way in order to provide a bit of margin/padding on the left/right side of the hint text and effectively "erase" the border; this is why it does not match your custom background. So let's assume you could set it to transparent, this is what your UI would look like:

![image](https://github.com/user-attachments/assets/405327c1-54f6-4792-baf5-894d1042a699)

The `HintAssist.Background` color is actually respected (it defaults to `Transparent` so you don't see it by default). However, in the special case of the outlined textbox where the hint needs to "erase" a border, the same AP `HintAssist.Background` is overridden by a trigger. I think perhaps this special case could have a dedicated AP (e.g. `HintAssist.OutlinedStyleFloatingBackground`) which could default to the window background color, but allow consumers to change if as they see fit (in your case you would change it to light blue). This would allow special handling for the special case.

In your case, if you set `HintAssist.Background=LightBlue` (assuming that is your background color), then you see the actual problem:

![image](https://github.com/user-attachments/assets/c0fbc779-ff94-4d4c-b972-42d2ffd8ac8f)

The background is respected, but there is an additional element (`Grid` named "HintBackgroundGrid") which is intended to add the margin/padding on the left/right of the hint (the white bars), and that STILL follows the window background. So with the current implementation, I don't believe you have a lot of options. I'll create a PR with my approach mentioned above and see what the other maintainers think about it.

**Comment by nicolaihenriksen** [comment]
@juusimaa @corvinsz I added this PR, https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/pull/3701,  which should resolve the issue.
